exporter/fileexporter/file_exporter.go

```diff
@@ -35,9 +35,10 @@ var marshaler = &jsonpb.Marshaler{}
 // fileExporter is the implementation of file exporter that writes telemetry data to a file
 // in Protobuf-JSON format.
 type fileExporter struct {
-	path  string
-	file  io.WriteCloser
-	mutex sync.Mutex
+	path         string
+	file         io.WriteCloser
+	mutex        sync.Mutex
+	isFileClosed bool
```

Comment:
This is not needed, we do guarantee in the SharedComponent that is called only once. So this can be reverted.

Reply:
Actually, I don't think so. I tried to make a fileexporter instance (with current master) shared in traces service and logs services. The `shutdown` was invoked twice.

Reply:
So do `start` functions.

Reply:
If you create via the factory it should not.

Reply:
See test that proves my point https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/defaultcomponents/default_exporters_test.go#L163. If I miss something let me konw.